ballot_picture/cli/main.go

```diff
@@ -2,22 +2,83 @@ package main
 
 import (
 	"flag"
+	"fmt"
+	"io/ioutil"
 	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/candidatos-info/enriquecedores/filestorage"
+	"github.com/matryer/try"
+)
+
+const (
+	maxAttempts = 5 // number of times to retry
 )
 
 func main() {
 	stateDir := flag.String("inDir", "", "diretório onde as fotos do estado estão")
-	candidaturesBucket := flag.String("candidatesDir", "", "bucket onde estão as as candidaturas")
-	picturesBucket := flag.String("outDir", "", "bucket onde as fotos devem ser salvas")
+	candidatesDir := flag.String("candidatesDir", "", "local onde estão as as candidaturas") // Se for para usar o gcs usar gs://BUCKET/ANO, se for local basta passar o path
+	picturesDir := flag.String("outDir", "", "local onde as fotos devem ser salvas")         // Se for para usar o gcs usar gs://BUCKET/ANO, se for local basta passar o path
 	flag.Parse()
 	if *stateDir == "" {
 		log.Fatal("informe o diretório onde as fotos do estão estão")
 	}
-	if *candidaturesBucket == "" {
-		log.Fatal("informe o bucket onde as candidaturas estão")
+	if *candidatesDir == "" {
+		log.Fatal("informe o local onde as candidaturas estão")
+	}
+	if *picturesDir == "" {
+		log.Fatal("informe o local onde as fotos devem ser salvs")
+	}
+	gcsClient, err := filestorage.NewGCSClient()
+	if err != nil {
+		log.Fatalf("falha ao criar cliente do Google Cloud Storage, erro %q", err)
+	}
+	logFileName := fmt.Sprintf("%s.txt", filepath.Base(*stateDir))
+	logErrorFile, err := os.Create(logFileName)
+	if err != nil {
+		log.Fatalf("falha ao criar arquivo de fotos com falha %s, erro %q", logFileName, err)
 	}
-	if *picturesBucket == "" {
-		log.Fatal("informe o bucket onde as fotos devem ser salvs")
+	defer logErrorFile.Close()
+	err = filepath.Walk(*stateDir, func(path string, info os.FileInfo, err error) error {
+		if path != *stateDir {
+			fileName := filepath.Base(path)
+			fileExtension := filepath.Ext(fileName)
+			sequencialCandidate := strings.TrimSuffix(fileName, fileExtension)
+			candidatureFilePath := fmt.Sprintf("%s.zip", sequencialCandidate)
+			if strings.HasPrefix(*candidatesDir, "gs://") { // using GCS
+				bucket := strings.ReplaceAll(*candidatesDir, "gs://", "")
```

Comment:
[sugestão] acho que a organização do código pode melhorar um pouco

1. checar se o arquivo de candidatos é GCS ou Local
2. checar se existe o arquivo lá (independente da opção acima do que for)
3. caso não exista, logar e continue no for
4. caso exista, checar se o picturesDir é local ou gcs
5. copiar